src/builder.rs

```diff
@@ -344,6 +346,18 @@ impl Builder {
         Ok(())
     }
 
+    /// Add an instruction to burn given amount os specific ZSA
+    pub fn add_burnt_asset(&mut self, asset: AssetId, value: i64) -> Result<(), &'static str> {
```

Comment:
This should be
```rust
pub fn add_burn(&mut self, asset: AssetId, value: NoteValue) -> Result<(), &'static str> {
```
And every addition of the value to the hash map should be tested for `overflow`.

Reply:
done